src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/MethodConfigurationComponent.scala

```diff
@@ -74,8 +77,155 @@ trait MethodConfigurationComponent {
     def configKeyIdx = index("IDX_MC_PREREQ", (methodConfigId, key), unique = true)
   }
 
-  protected val methodConfigurationQuery = TableQuery[MethodConfigurationTable]
   protected val methodConfigurationInputQuery = TableQuery[MethodConfigurationInputTable]
   protected val methodConfigurationOutputQuery = TableQuery[MethodConfigurationOutputTable]
   protected val methodConfigurationPrereqQuery = TableQuery[MethodConfigurationPrereqTable]
+
+  object methodConfigurationQuery extends TableQuery(new MethodConfigurationTable(_)) {
+
+    private type MethodConfigurationQueryType = driver.api.Query[MethodConfigurationTable, MethodConfigurationRecord, Seq]
+    private type MethodConfigurationInputQueryType = driver.api.Query[MethodConfigurationInputTable, MethodConfigurationInputRecord, Seq]
+    private type MethodConfigurationOutputQueryType = driver.api.Query[MethodConfigurationOutputTable, MethodConfigurationOutputRecord, Seq]
+    private type MethodConfigurationPrereqQueryType = driver.api.Query[MethodConfigurationPrereqTable, MethodConfigurationPrereqRecord, Seq]
+
+    /*
+      the core methods
+     */
+    def save(workspaceId: UUID, methodConfig: MethodConfiguration): ReadWriteAction[MethodConfiguration] = {
+
+      def saveMaps(configId: Long) = {
+        DBIO.seq(methodConfig.prerequisites.map { case (key, value) =>
+          (methodConfigurationPrereqQuery returning methodConfigurationPrereqQuery.map(_.id)) += MethodConfigurationPrereqRecord(configId, 0, key, value.value)
+        }.toSeq: _*) andThen
+        DBIO.seq(methodConfig.inputs.map { case (key, value) =>
+          (methodConfigurationInputQuery returning methodConfigurationInputQuery.map(_.id)) += MethodConfigurationInputRecord(configId, 0, key, value.value)
+        }.toSeq: _*) andThen
+        DBIO.seq(methodConfig.outputs.map { case (key, value) =>
+          (methodConfigurationOutputQuery returning methodConfigurationOutputQuery.map(_.id)) += MethodConfigurationOutputRecord(configId, 0, key, value.value)
+        }.toSeq: _*)
+      }
+
+      uniqueResult[MethodConfigurationRecord](findConfigByName(workspaceId, methodConfig.namespace, methodConfig.name)) flatMap {
+        case None =>
+          val configInsert = (methodConfigurationQuery returning methodConfigurationQuery.map(_.id)) += MethodConfigurationRecord(0, methodConfig.namespace, methodConfig.name, workspaceId, methodConfig.rootEntityType, methodConfig.methodRepoMethod.methodNamespace, methodConfig.methodRepoMethod.methodName, methodConfig.methodRepoMethod.methodVersion)
+          configInsert flatMap { configId =>
+            saveMaps(configId)
+          }
+        //is this the best approach? delete all of the inputs/outputs/prereqs and then save the new ones
+        case Some(methodConfigRec) =>
```

Comment:
this won't update the root entity type if that's changed when a config is re-saved. will fix